rollup/internal/orm/blob_upload.go

```diff
@@ -36,23 +37,87 @@ func (*BlobUpload) TableName() string {
 	return "blob_upload"
 }
 
+// GetNextBatchIndexToUploadByPlatform retrieves the next batch index that hasn't been uploaded to corresponding blob storage service
+func (o *BlobUpload) GetNextBatchIndexToUploadByPlatform(ctx context.Context, startBatch uint64, platform types.BlobStoragePlatform) (uint64, error) {
+	db := o.db.WithContext(ctx)
+	db = db.Model(&BlobUpload{})
+	db = db.Where("platform = ? AND status = ? AND deleted_at IS NULL", platform, types.BlobUploadStatusUploaded)
```

Comment:
don't need to add `AND deleted_at IS NULL`, gorm will automatically generate it.